lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -1530,19 +1523,21 @@ struct ConvertLayoutOpConversion
     for (unsigned repId = 0; repId < accumNumReplicates; ++repId) {
       auto multiDimRepId = getMultiDimIndex<unsigned>(repId, numReplicates);
       rewriter.create<mlir::gpu::BarrierOp>(loc);
-      if (auto srcBlockedLayout = srcLayout.dyn_cast<BlockedEncodingAttr>()) {
-        processReplicaBlocked(loc, rewriter, /*stNotRd*/ true, srcTy,
-                              inNumCTAsEachRep, multiDimRepId, inVec,
-                              paddedRepShape, outOrd, vals, smemBase);
+      if (srcLayout.isa<BlockedEncodingAttr>() ||
```

Comment:
In the future, maybe more layouts will be added, or the compatibility semantics of layout conversion may change, so it may be better to put the layout validity check as a member method into Layout class itself, rather than directly put it in this level, thus making it easier for future change.

Reply:
Not sure if I fully get your point. I think one thing that could be improved is the derivative relationships between these layout Attributes. It seems to me that tablegen does not support derivative relationship between two customized Attrs, and that brings some ugly codes. Haven't got time to further investigate anyway.

Reply:
Got it, okay for me now.